File: ComponentDemo/public_html/fileExplorer/jsLibs/ratingConverter.js
/*
** Copyright (c) 2008, 2014, Oracle and/or its affiliates. All rights reserved.
*/
function RatingConverter()
{
  // for debugging
  this._class = "RatingConverter";
}

RatingConverter.prototype = new TrConverter();

RatingConverter.prototype.getFormatHint = function()
{
  return null;
}

RatingConverter.prototype.getAsString = function(number, label)
{
  var numberString = "" + number;
  
  if(numberString=="0" || numberString == "0.0")
  {
    return "Needs Significant Improvement";
  }
  else if(numberString=="1" || numberString == "1.0")
  {
    return "1";
  }
  else if(numberString=="2" || numberString == "2.0")
  {
    return "2";
  }
  else if(numberString=="3" || numberString == "3.0")
  {
    return "3";
  }
  else if(numberString=="4" || numberString == "4.0")
  {
    return "4";
  }
  else if(numberString=="5" || numberString == "5.0")
  {	
    return "Average";
  }
  else if(numberString=="6" || numberString == "6.0")
  {
    return "6";
  }
  else if(numberString=="7" || numberString == "7.0")
  {
    return "7";
  }
  else if(numberString=="8" || numberString == "8.0")
  {
    return "8";
  }
  else if(numberString=="9" || numberString == "9.0")
  {
    return "9";
  }
  else if(numberString=="10" || numberString == "10.0")
  {
    return "The Best!";
  }
  return numberString;
}

RatingConverter.prototype.getAsObject = function(numberString, label)
{
  if(numberString=="Needs Significant Improvement")
  {
    return 0;
  }
  else if(numberString=="1")
  {
    return 1;
  }
  else if(numberString=="2")
  {
    return 2;
  }
  else if(numberString=="3")
  {
    return 3;
  }
  else if(numberString=="4")
  {
    return 4;
  }
  else if(numberString=="Average")
  {
    return 5;
  }
  else if(numberString=="6")
  {
    return 6;
  }
  else if(numberString=="7")
  {
    return 7;
  }
  else if(numberString=="8")
  {
    return 8;
  }
  else if(numberString=="9")
  {
    return 9;
  }	
  else if(numberString=="The Best!")
  {
    return 10;
  }

  return numberString;
}

File: ComponentDemo/src/oracle/adfdemo/view/explorer/rich/converter/RatingConverter.java
/* Copyright (c) 2008, 2014, Oracle and/or its affiliates. 
All rights reserved.*/

package oracle.adfdemo.view.explorer.rich.converter;

import java.util.Collection;

import javax.faces.component.UIComponent;
import javax.faces.context.FacesContext;
import javax.faces.convert.Converter;

import org.apache.myfaces.trinidad.convert.ClientConverter;

public class RatingConverter implements Converter, ClientConverter 
{
  public Object getAsObject(FacesContext arg0, UIComponent arg1, String arg2)
  {
    Double d = null;
    if ("Needs Significant Improvement".equals(arg2))
      d = new Double(0);
    else if ("1".equals(arg2))
      d = new Double(1);
    else if ("2".equals(arg2))
      d = new Double(2);
    else if ("3".equals(arg2))
      d = new Double(3);
    else if ("4".equals(arg2))
      d = new Double(4);
    else if ("Average".equals(arg2))
      d = new Double(5);
    else if ("6".equals(arg2))
      d = new Double(6);
    else if ("7".equals(arg2))
      d = new Double(7);
    else if ("8".equals(arg2))
      d = new Double(8);
    else if ("9".equals(arg2))
      d = new Double(9);
    else if ("The Best!".equals(arg2))
      d = new Double(10);
    else
      d = new Double(arg2);
    return d;
  }

  public String getAsString(FacesContext arg0, UIComponent arg1, Object arg2) 
  {
    if (arg2 instanceof Double) 
    {
      String dou = ((Double)arg2).toString();
      if (dou.equals("0.0"))
        dou = "Needs Significant Improvement";
      if (dou.equals("1.0"))
        dou = "1";
      else if (dou.equals("2.0"))
        dou = "2";
      else if (dou.equals("3.0"))
        dou = "3";
      else if (dou.equals("4.0"))
        dou = "4";
      else if (dou.equals("5.0"))
        dou = "Average";
      else if (dou.equals("6.0"))
        dou = "6";
      else if (dou.equals("7.0"))
        dou = "7";
      else if (dou.equals("8.0"))
        dou = "8";
      else if (dou.equals("9.0"))
        dou = "9";
      else if (dou.equals("10.0"))
        dou = "The Best!";
      return dou;
    } 
    else
    {
      return arg2.toString();
    }
  }

  public String getClientConversion(FacesContext arg0, UIComponent arg1) 
  {
    return ("new RatingConverter()");
  }

  public Collection<String> getClientImportNames() 
  {
    return null;
  }

  public String getClientLibrarySource(FacesContext context) 
  {
    return context.getExternalContext().getRequestContextPath() + 
           "/fileExplorer/jsLibs/ratingConverter.js";
  }

  public String getClientScript(FacesContext arg0, UIComponent arg1) 
  {
    return null;
  }
}
